secure_message/services/party_service.py

```diff
@@ -1,49 +1,53 @@
-
-    def get_business_details(self, ru):
+    @staticmethod
+    def get_business_details(ru_ids):
         """Retrieves the business details from the party service"""
-
-        ru_dict = self._business_details_cache.get(ru)
-
-        if ru_dict is None:
-            logger.info(f"Party Service: retrieve party ru data for {ru}")
-
-            party_data = requests.get(f"{current_app.config['RAS_PARTY_SERVICE']}party-api/v1/businesses/id/{ru}",
-                                      auth=current_app.config['BASIC_AUTH'], verify=False)
-            if party_data.status_code == 200:
-                ru_dict = party_data.json()
-                self._business_details_cache[ru] = ru_dict
-                logger.debug(f"Party data retrieved for ru:{ru}")
-            else:
-                logger.debug(f"Party data failed for ru:{ru}", status_code=party_data.status_code, text=party_data.text,
-                             ru=ru)
-
-        return ru_dict
+        payload = urlencode([("id", ru_id) for ru_id in ru_ids])
+        response = requests.get(f"{current_app.config['RAS_PARTY_SERVICE']}party-api/v1/businesses",
+                                auth=current_app.config['BASIC_AUTH'], verify=False, params=payload)
+        try:
+            response.raise_for_status()
+        except HTTPError:
+            logger.debug("Business detail retrieval failed", status_code=response.status_code, text=response.text,
+                         ru_ids=ru_ids)
+            return []
+
+        logger.debug("Business details sucessfully retrieved", ru_ids=ru_ids)
+        return response.json()
 
     def get_user_details(self, uuid):
-
-        user_dict = self._users_cache.get(uuid)
-
-        if user_dict is None:
-            logger.info(f"Party Service: retrieve party user data for {uuid}")
-            party_data = requests.get(f"{current_app.config['RAS_PARTY_SERVICE']}party-api/v1/respondents/id/{uuid}",
-                                      auth=current_app.config['BASIC_AUTH'], verify=False)
-            if party_data.status_code == 200:
-                user_dict = party_data.json()
-                self._users_cache[uuid] = user_dict
-                logger.debug(f"Party data retrieved for uuid:{uuid}")
-            else:
-                logger.error(f'Party service failed for uuid:{uuid}', status_code=party_data.status_code, text=party_data.text,
-                             uuid=uuid)
-        return user_dict
+        """Retrieves the user details from the party service after converting single
+        uuid into a list of one"""
+        return self._get_user_details_from_party_service([uuid])
+
+    def get_users_details(self, uuids):
+        """Retrieves the user details from the party service"""
+        return self._get_user_details_from_party_service(uuids)
+
+    @staticmethod
+    def _get_user_details_from_party_service(uuids):
+        payload = urlencode([("id", uuid) for uuid in uuids])
```

Comment:
I think `payload` is misleading, it makes me think of something we'd be posting, maybe `id_params`?

Reply:
Fixed